### ISSUE TITLE
Rename configuration directory to reference project's name

### ISSUE THREAD

**Issue by andreyorst** [clojure-lsp/clojure-lsp#476]
Right now `clojure-lsp` uses `.lsp` directory for storing server configuration file. This has several problems:

1. `.lsp` is overly-generic name, other language servers may use it.
2. `~/.lsp` and `~/.config/.lsp` doesn't tell what exactly is being configured - is it a general settings for all language servers or config for a particular client, or a server config, and if so what server?
3. `~/.config/.lsp` is an unusual approach of storing configuration in a hidden directory. `~/.config` is already hidden by itself, and it's contents are meant to be visible.
4. `.lsp` works fine for project local configuration, because no other language server will likely be used in a Clojure project, but this is still possible, when working on a multi-language project with Java. It is unlikely that any other server will look for `config.edn` though.

I would like to propose a change from `~/.lsp` to `~/.clojure-lsp` and from `$XDG_CONFIG_HOME/.lsp` to `$XDG_CONFIG_HOME/clojure-lsp`. We can support both styles for some releases, showing warning in a log that `.lsp` is deprecated.

On project local configuration I'm more open to keep `.lsp` but for consistency sake we might want to use `.clojure-lsp` as well. This would keep implementation code cleaner.

This change requires modifying every reference on the docs.

**Comment by ericdallo** [comment]
Just linking my opinion on this [here](https://github.com/clojure-lsp/clojure-lsp/issues/474#issuecomment-878530175).

TLDR: agree with all points proposed, so we should keep the old .lsp for a while displaying a warning message on server log.

c/c @snoe if see any issues with this approach.

**Comment by andreyorst** [comment]
I've recently updated the server, and noticed that some of my configurations weren't applying correctly. Then I've referred to https://clojure-lsp.io/settings/#global and noticed that the name of the config directory changed from `lsp` to `clojure-lsp`, but I wasn't able to find it in the changelog. I've renamed the directory and now the settings work again, so it's not a problem for me, just seems like a undocumented potentially breaking change. Is it a correct behavior that `~/.config/lsp` is ignored right now and not considered deprecated?

**Comment by ericdallo** [comment]
I think so @andreyorst, probably a breaking change we didn't notice in some refactor, we only look for:
- ~/.config/clojure-lsp
- ~/.lsp/ (as backward compatibility fallback)

**Comment by snoe** [comment]
For global user config, this sounds fine. A warning in the logs probably won't reach people so a message to the clients might be better. An alternative to a warning is to just move the config to the new location (ask first). 

For project config, the only concern is that these files are generally checked in on teams where not everyone may be on the same version of the server. A lot of caches will need to be recalculated if it's moved.

**Comment by ericdallo** [comment]
also, multiple gitignores already exists and templates which would lost track of this :(

**Comment by ericdallo** [comment]
I still think having project config as .clojure-lsp but change that is something a little bit critical, so I will keep this open until we think in a better way, maybe just support the old way and update docs
